JSON path traversing raises exception rather than not using the route

Thanks for developing and documenting repsx, it is extremely useful.

One such useful features is the json pattern path traversing.
However I am having an issue that if I don't have a matching pattern before one where I use path traversing I get an error if my request has json but not that key.

```python
#conftest.py
@pytest.fixture(scope="module")
def mock_auth_api():
    with respx.mock(base_url="https://login.example.com/") as respx_mock:
        example_route = respx_mock.post(
            path__regex=r"([0-9a-zA-Z-]+)/oauth2/token$",
            json__grant_type="refresh_token",
        ).mock(side_effect=my_side_effect)
        yield respx_mock

#example_test.py
request_url = "https://app.diffhost.net/items/hFj8G5/endpoint"
headers = {"Accept": "*/*", "Authorization": "Bearer 123" }
json = {"type": "example"}
response = httpx.post(request_url, headers=headers, json=body)
```

Result:

```console
self = <JSON eq '"refresh_token"'>, request = <Request('POST', 'https://app.diffhost.net/items/hFj8G5/endpoint')>

    def parse(self, request: httpx.Request) -> str:
        content = super().parse(request)
        try:
            json = jsonlib.loads(content.decode("utf-8"))
        except JSONDecodeError:
            return ""
    
        if self.path:
            value = json
            for bit in self.path.split("__"):
                key = int(bit) if bit.isdigit() else bit
                try:
                    value = value[key]
                except KeyError as e:
>                   raise KeyError(f"{self.path!r} not in {json!r}") from e
E                   KeyError: "'grant_type' not in {'type': 'example'}"
```

**Expected:**

I would expect the KeyError to result in a non-match pattern, not an exception being raised.

Additionally it seems odd to me that since the base_url and path both do not match that the rest of the pattern is even checked. I presume using `pattern = M(path...) & M(json...)` would not change this?

--- a/respx/patterns.py
+++ b/respx/patterns.py
@@ -123,7 +123,10 @@ class Pattern(ABC):
         return value
 
     def match(self, request: httpx.Request) -> Match:
-        value = self.parse(request)
+        try:
+            value = self.parse(request)
+        except Exception:
+            return Match(False)
 
         # Match and strip base
         if self.base:

--- a/tests/test_patterns.py
+++ b/tests/test_patterns.py
@@ -338,21 +338,15 @@ def test_json_pattern(lookup, value, json, expected):
         ({"ham": [{"spam": "spam"}, {"egg": "yolk"}]}, "ham__1__egg", "yolk", True),
         ([{"name": "jonas"}], "0__name", "jonas", True),
         ({"pk": 123}, "pk", 123, True),
-        ({"foo": {"bar": "baz"}}, "foo__ham", "spam", KeyError),
-        ([{"name": "lundberg"}], "1__name", "lundberg", IndexError),
+        ({"foo": {"bar": "baz"}}, "foo__ham", "spam", False),
+        ([{"name": "lundberg"}], "1__name", "lundberg", False),
     ],
 )
 def test_json_pattern_path(json, path, value, expected):
     request = httpx.Request("POST", "https://foo.bar/", json=json)
     pattern = M(**{f"json__{path}": value})
-    if type(expected) is bool:
-        match = pattern.match(request)
-        assert bool(match) is expected
-    elif issubclass(expected, Exception):
-        with pytest.raises(expected):
-            pattern.match(request)
-    else:
-        raise AssertionError()  # pragma: nocover
+    match = pattern.match(request)
+    assert bool(match) is expected
 
 
 def test_invalid_pattern():

